### PR TITLE
Fix broken default logo image

### LINKED ISSUE #3656: Default logo image not properly saved
### CKAN Version if known (or site URL)
2.7 and 2.8.0a

By default, CKAN uses a default logo image. This can be changed through the UI in the sysadmin config page or in configuration settings.

If you go to the sysadmin config page, and if you don't change the logo image and just click Update, the logo image will break. The reason is because its been given wrong path to the image.

### What steps can be taken to reproduce the issue? 
Go to sysadmin config page. If you previously changed the default logo image, click the Reset button to bring the default settings. After that click on Update and the logo image will break.

I isolated the issue and will provide a fix.

### PR DESCRIPTION
**Pull request by klikstermkd**
Fixes #3656 

### Features:

- [ ] includes tests covering changes
- [ ] includes updated documentation
- [ ] includes user-visible changes
- [ ] includes API changes
- [X] includes bugfix for possible backport

### CODE REVIEW

**Comment by klikstermkd** [comment]
Ready for review.

File: ckan/logic/action/update.py [hunk 1]
```diff
@@ -1270,11 +1271,27 @@ def config_option_update(context, data_dict):
...
     for key, value in data.iteritems():
 
         # Set full Logo url
-        if key =='ckan.site_logo' and value and not value.startswith('http'):
-            value = h.url_for_static('uploads/admin/{0}'.format(value))
+        if key == 'ckan.site_logo' and value and not value.startswith('http'):
+            for f in os.listdir(ckan_images_path):
+                if f == value:
+                    image_in_ckan = True
+                    break
```
Comment (wardi) [inline ckan/logic/action/update.py]: os.isfile can be used instead of a for loop here

**Comment by klikstermkd** [comment]
@wardi Now it should be good.

**Comment by klikstermkd** [comment]
@wardi Can you check this again? :)

**Comment by amercader** [comment]
I think the underlying problem is that the admin form is receiving the correct value (`/base/images/ckan-logo.png`), even if the widget just displays the file name:

```html
<input id="field-image-url" type="url" name="ckan.site_logo" value="/base/images/ckan-logo.png" placeholder="http://example.com/my-image.jpg" class="form-control" readonly="">
```

But sending the wrong value to the controllers, it will only send the file name (`ckan-logo.png`). Rather than adapt the action to this, it would be best to ensure that the form is sending data unchanged.

**Comment by klikstermkd** [comment]
That still won't resolve the issue because the controller doesn't know whether the default image is uploaded from a user or available already from CKAN's local images directory.

**Comment by amercader** [comment]
True, but then it's just a matter of covering the local image case in the `config_option_update()` check:

```python
if key =='ckan.site_logo' and value and not (value.startswith('http') or value.startswith('/')):
```

**Comment by klikstermkd** [comment]
Uploaded images from user and CKAN's images from local folder will pass the condition.

**Comment by amercader** [comment]
Sorry, I meant: 

```python
if key =='ckan.site_logo' and value and not value.startswith('http') and not value.startswith('/'):
```

**Comment by klikstermkd** [comment]
@amercader Can you check now?

**Pull request merged**